src/prefect/engine/task_runner.py

```diff
@@ -893,28 +889,27 @@ def get_task_run_state(
         except signals.LOOP as exc:
             new_state = exc.state
             assert isinstance(new_state, Looped)
-            new_state.result = Result(
-                value=new_state.result, result_handler=self.result_handler
+            new_state.result = self.task.result.write(
+                new_state.result, **prefect.context
```

Comment:
I think this is taking advantage of the fake that `new_state.result`, as opposed to `new_state._result`, peels out `new_state._result.value` here, but it was pretty weird reading so want to double check that. It does make this call "look" different from all the other ones.